tests/subsys/logging/log_syst/src/mock_backend.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "mock_backend.h"
+#include <ztest.h>
+#include <logging/log_core.h>
+#include <logging/log_backend_std.h>
+#include <stdlib.h>
+
+static uint32_t log_format_current = CONFIG_LOG_BACKEND_MOCK_OUTPUT_DEFAULT;
+union log_msg2_generic *test_msg;
+
+static uint8_t mock_output_buf[1];
+uint8_t test_output_buf[256];
+int i;
+
+/**
+ * @brief This function takes in the characters and copies
+ * them in the temporary buffer.
+ *
+ * @return No. of characters copied to the temporary buffer.
+ */
+static int char_out(uint8_t *data, size_t length, void *ctx)
+{
+	ARG_UNUSED(ctx);
+	size_t output_length = length;
+
+	while (length > 0) {
+		test_output_buf[i] = *data;
+		i++;
+		data++;
+		length--;
+
+		if (i == sizeof(test_output_buf)-1) {
+			i = 0;
```

Comment:
Is it the behavior you want ? Overwrite the buffer data is bigger than the space you have.

Reply:
When the buffer is full, the buffer size needs to be increased. In our case this block would not have executed since we know the input strings passed from the testcase. But good point, replaced this logic with zassert statement which would ask to increase the size of buffer when it is full instead of overriding it.